### PR TITLE
1559 JSON RPC Diff

### PR DESCRIPTION
**Pull request by alita-moore**
(sorry about the bad commit name)
This..
- adds `baseFee` to the `Block` type
- removes `gasPrice` from `Transaction` type
- adds `maxPriorityFeePerGas` to `Transaction` type
- adds `maxFeePerGas` to `Transaction` type

### CODE REVIEW

**Review by MicahZoltu — APPROVED** [review]

File: json-rpc/spec.json [hunk 1]
```diff
@@ -1464,6 +1464,11 @@
               "description": "Block hash of the RLP encoding of an uncle block",
               "$ref": "#/components/schemas/Keccak"
             }
+          },
+          "baseFee": {
+            "title": "baseFee",
+            "description": "The block's baseline network fee per gas; the baseFee is burned",
+            "$ref": "#/components/schemas/Integer"
           }
```
Comment (MicahZoltu) [inline json-rpc/spec.json:1472]: This needs to be optional.

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
We need to make it clear that it is not valid to have both `gasPrice` and `maxPriorityFeePerGas` and other similar combinations.

File: json-rpc/spec.json [hunk 2]
```diff
@@ -1464,6 +1464,11 @@
               "description": "Block hash of the RLP encoding of an uncle block",
               "$ref": "#/components/schemas/Keccak"
             }
+          },
+          "baseFee": {
+            "title": "baseFee",
+            "description": "The block's baseline network fee per gas; the baseFee is burned",
+            "$ref": "#/components/schemas/Integer"
```
[the remark this answers is not in the document]
Comment (alita-moore) [reply json-rpc/spec.json:1471]: good point, I can add that. Although goes a little beyond the scope of 1559 changes. As I understand it this was copy pasted from ETC so I'll be doing some conditioning like you mention soon

**Comment by timbeiko** [comment]
@alita-moore FYI we agreed on ACD this AM to use the terms from the EIP, so `baseFeePerGas`, `maxFeePerGas`, `maxPriorityFeePerGas`.

File: json-rpc/spec.json [hunk 3]
```diff
@@ -1464,6 +1464,11 @@
               "description": "Block hash of the RLP encoding of an uncle block",
               "$ref": "#/components/schemas/Keccak"
             }
+          },
+          "baseFee": {
+            "title": "baseFee",
```
Comment (timbeiko) [inline json-rpc/spec.json:1469]:
```suggestion
            "title": "baseFeePerGas",
```

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
Would you mind splitting the spec changes from the spec rename?  Sadly, GitHub doesn't seem to recognize this is a file rename so we can't see the "diff", which is valuable from a history standpoint.

I recommend doing the file move *first*, so that change history is easily traversed.

**Comment by holgerd77** [comment]
What is actually the reason for this renaming respectively is this really necessary? Will there be more specs in different formats in the future?

In case renaming is wanted short note that there is a link in the main README.md which then would also need an update.

**Comment by lightclient** [comment]
I agree, I don't really see a reason to change until there are more spec formats.

**Comment by alita-moore** [comment]
@holgerd77 the reason is because the [OPEN-RPC vscode plugin](https://marketplace.visualstudio.com/items?itemName=OPEN-RPC.OPEN-RPC) uses the `openrpc.json` as a way to find the files to parse. It's not required of course, but it is helpful when making changes. Either way I have reverted the change

**Comment by timbeiko** [comment]
@alita-moore @lightclient @MicahZoltu is this OK to merge now?

**Comment by alita-moore** [comment]
@MicahZoltu just a heads I'm going to address your comment about the integer in a separate PR

**Comment by MicahZoltu** [comment]
If we are going to rename this (and I think @alita-moore's argument for renaming is quite reasonable) we should do it *before* this PR so history doesn't get confused by the rename.

**Comment by alita-moore** [comment]
@MicahZoltu i will be making significant changes to the structure of the spec. I thinks it's best to just do it after. There is no practical difference in my opinion

File: json-rpc/spec.json [hunk 4]
```diff
@@ -1497,76 +1523,112 @@
...
+            },
+            {
+                "oneOf": [
+                    {
+                        "title": "Block <1559> to PRESENT",
+                        "type": "object",
+                        "properties": {
+                            "maxPriorityFeePerGas": {
+                                "title": "maxPriorityFeePerGas",
+                                "description": "maximum fee per gas the sender is willing to pay to miners in wei",
```
Comment (lightclient) [inline json-rpc/spec.json:1608]:
```suggestion
                                "description": "Maximum fee per gas the sender is willing to pay to miners in wei",
```

**Review by lightclient — APPROVED** [review]
looks correct to me

File: json-rpc/spec.json [hunk 5]
```diff
@@ -1497,76 +1523,112 @@
...
+                                "$ref": "#/components/schemas/Integer"
+                            },
+                            "maxFeePerGas": {
+                                "title": "maxPriorityFeePerGas",
```
Comment (lightclient) [inline json-rpc/spec.json:1612]:
```suggestion
                                "title": "maxFeePerGas",
```

**Review by lightclient — CHANGES_REQUESTED** [review]
Just caught an issue actually.

File: json-rpc/spec.json (hunk 3 above)
Comment (ShayneMichaelBacik123) [reply json-rpc/spec.json]: Payment is required.donations to autodeposit email:shaynemichaelhall@gmail.com

**Pull request merged**